files/image_config/systemd/systemd-sonic-generator.j2

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/python
+
+import os
+import sys
+
+UNIT_FILE_PREFIX = '/etc/systemd/system/'
+
+def parse_install_targets(unit_file):
+    '''Return installation information for a unit file.'''
+
+    install_targets_raw = [] 
+
+    install_targets = []
+    
+    with open(os.path.join(UNIT_FILE_PREFIX, unit_file), 'r') as f:
+        found_install = False
+        for line in f:
+            if '[Install]' in line:
+                found_install = True
+            elif found_install:
+                install_targets_raw.append(line)
+    
+    for line in install_targets_raw:
+        install_option = line.split('=')[0]
+        targets = line.split('=')[1].split()
+        if 'WantedBy' in install_option:
+            for target in targets:
+                install_targets.append(target + '.wants') 
+        elif 'RequiredBy' in install_option:
+            for target in targets:
+                install_targets.append(target + '.requires')
+
+    return install_targets
+             
+ 
+def install_unit(unit_file, install_dir, targets, instance = None):
+    '''Create symlinks for a unit file installation'''
+     
+    for target in targets:
+        if instance:
+            # Insert the instance into the symlink destination
+            tmp = unit_file.split('@')
+            unit_file_instance = '{}@{}{}'.format(tmp[0], instance, tmp[1])
+            dst = os.path.join(install_dir, target, unit_file_instance)
+        else:
+            dst = os.path.join(install_dir, target, unit_file)
+        src = os.path.join(UNIT_FILE_PREFIX, unit_file)
+        os.symlink(src, dst)
+
+
+def main(install_dir):
+    standalone_services = "ntp-config.service warmboot-finalizer.service rsyslog-config.service hostcnfgd.service hostname-config.service updategraph.service caclmgrd.service process-reboot-cause.service fstrim.timer snmp.timer".split()
+
+    for service in standalone_services:
+        targets = parse_install_targets(service)
+        install_unit(service, install_dir, targets)
```

Comment:
do we need to put standalone service into the generator.

Reply:
I feel like it's better to have everything in one place, plus it will be easier if we ever need to enable/disable these based on platform.

Reply:
At the very least rsyslog-config and updategraph belong here since they will need to be enabled differently depending on platform (in a future PR)

Reply:
we should only put the services that needs generator in the generator function, the services that do not have instance should not be put into the generator function such as standalone services.

Reply:
Ok, I'll make the change.

Reply:
@lguohan on second thought I think all the services need to be put inside the generator since their dependencies will change based on whether or not a multi-asic platform is detected, i.e. ntp-config.service is dependent on updategraph.service, but for a multi-asic platform it will be dependent on updategraph@0.service, updategraph@1.service, etc.

Reply:
Just curious why we will have multiple instances of services like updategraph. Even with multiple ASICS, we would still load only one minigraph for the entire device, correct? I would expect that even on a multi-ASIC device, all ASIC-specific services would all depend on one instance of updategraph (as well as onloy one rsyslog-config, ntp-config, etc.). Is there some limitation I'm not aware of?